apps/server/src/modules/h5p-editor/service/libraryStorage.service.ts

```diff
@@ -311,20 +305,11 @@ export class LibraryStorage implements ILibraryStorage {
 	 * @param library
 	 */
 	public async getLanguages(libraryName: ILibraryName): Promise<string[]> {
-		const library = await this.libraryRepo.findOneByNameAndVersionOrFail(
-			libraryName.machineName,
-			libraryName.majorVersion,
-			libraryName.minorVersion
-		);
+		const prefix = this.getS3Key(libraryName, 'language');
 
-		const languages: string[] = [];
-		for (const file of library.files) {
-			if (file.name.startsWith('language') && file.name.endsWith('.json')) {
-				languages.push(path.basename(file.name, '.json'));
-			}
-		}
+		const files = await this.s3Client.list(prefix);
 
-		return languages;
+		return files.filter((file) => path.extname(file) === '.json').map((file) => path.basename(file, '.json'));
```

Comment:
see above (especially this is very hard to read)

Reply:
Refactored for better readability